### PR TITLE
🔐 Add Sha256sum verification for GeoSite db + other fixes and cleanup

### PR DESCRIPTION
**Pull request by database64128**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [ ] Require translation update
- [ ] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New feature

---

### Description of your *pull request* and other information

- Using `System.Net.Http.HttpClient` for `GeositeUpdater`.
- Verifies downloaded geosite db by comparing sha256sum before committing the change.
- Removed some legacy compatibility code from `Program.cs` as we are not using .NET Framework.
- Changed `CurrentDirectory` detection method so that Self-Contained .exe can have its configuration files in the same directory as the .exe instead of in the temporary folder.
- Added some PublishProfiles for convenience.
- Updated dependencies to their latest versions.
- Code cleanup.

### CODE REVIEW

**Comment by chenshaoju** [comment]
Thanks for sending the pull request, Please waiting for the developer to review the code.

File: shadowsocks-csharp/Controller/Service/GeositeUpdater.cs [hunk 1]
```diff
@@ -73,31 +93,83 @@ public static void UpdatePACFromGeosite(Configuration config)
...
+
+                // verify sha256sum
+                byte[] downloadedDBHashBytes = mySHA256.ComputeHash(downloadedBytes);
+                string downloadedDBHash = BitConverter.ToString(downloadedDBHashBytes).Replace("-", String.Empty);
+                logger.Info($"Actual Sha256sum: {downloadedDBHash}");
+                if (geositeSha256sum != downloadedDBHash)
                 {
-                    File.WriteAllBytes(DATABASE_PATH, e.Result);
-                    LoadGeositeList();
+                    logger.Info("Sha256sum mismatch. Updating aborted.");
```
Comment (celeron533) [inline shadowsocks-csharp/Controller/Service/GeositeUpdater.cs:138]: "Sha256 checksum mismatch, file may corrupted. Abort the update."

File: shadowsocks-csharp/Controller/Service/GeositeUpdater.cs [hunk 2]
```diff
@@ -73,31 +93,83 @@ public static void UpdatePACFromGeosite(Configuration config)
...
+                    throw new Exception("Sha256sum mismatch");
+                }
+                else
+                {
+                    logger.Info("Sha256sum verification successful.");
```
Comment (celeron533) [inline shadowsocks-csharp/Controller/Service/GeositeUpdater.cs:143]: "Sha256sum verification: PASS"

File: shadowsocks-csharp/Controller/Service/GeositeUpdater.cs (hunk 1 above)
Comment (celeron533) [inline shadowsocks-csharp/Controller/Service/GeositeUpdater.cs:138]: "Sha256sum verification: FAIL. File is corrupted. Abort the update."

Comment (database64128) [reply shadowsocks-csharp/Controller/Service/GeositeUpdater.cs]: 👌 Refined the logging messages in b3e15fc.

**Pull request merged**